protocol/0087-EVMD-eth-rpc-and-evm-data-source.md

```diff
@@ -0,0 +1,135 @@
+# Ethereum RPC and EVM based data sources
+
+## Summary
+
+This specification adds a new way of sourcing data from any chain or Ethereum Layer 2 blockchain (L2) that supports Ethereum RPC calls and runs an EVM.
+
+> [!TIP]
+> A Layer 2 blockchain refers to network protocols that are layered on top of a Layer 1 solution. Layer 2 protocols use the Layer 1 blockchain for network and security infrastructure.
+
+The data is to be ingested as an Ethereum [data source](./0045-DSRC-data_sourcing.md) but pointing at a different RPC endpoint.
+Hence this is in addition to and building upon [Ethereum data source](./0082-ETHD-ethereum-data-source.md).
+
+## Description
+
+In addition to listening to Ethereum events and reading from Ethereum contracts as described in [Ethereum data source](./0082-ETHD-ethereum-data-source.md) it will be possible for Vega nodes to listen to events from and read from other chains that implement Ethereum RPC and run EVM, in particular Ethereum L2.
+
+The overarching principle is that the chain provides ethereum RPC / EVMs and thus contracts and ABIs are assumed to be functionally the same as on Ethereum itself.
+
+## Registration / removal
+
+A new network parameter, a JSON list of network id / chain id / name one per Ethereum RPC and EVM chain will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md).
+Name `blockchains.ethereumL2Configs`.
+Example value:
+
+```json
+{"configs": [{"network_id": 10, "chain_id": 10, "confirmations": 3, "name":"optimism"}]}
+```
+
+Duplicate values of `network_id`, `chain_id` or  `name` are not allowed (an update will be rejected at validation stage).
+Any update must always change the entire JSON (it's not possible to change individual entries).
```

Comment:
Only number of confirmations can be "changed", any other change is "removal" + "addition" in effect.

Reply:
In current minimal implementation, at proposal validation, check that only change is a) changing number of confirmation or b) adding another source. Add ACs to that effect.